go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -117,74 +114,84 @@ func (tm *TabletManager) restoreDataLocked(ctx context.Context, logger logutil.L
 		StartTime:           logutil.ProtoToTime(keyspaceInfo.SnapshotTime),
 	}
 
-	// Loop until a backup exists, unless we were told to give up immediately.
-	var backupManifest *mysqlctl.BackupManifest
-	for {
-		backupManifest, err = mysqlctl.Restore(ctx, params)
-		if waitForBackupInterval == 0 {
-			break
+	if mysqlctl.ShouldRestore(ctx, params) {
```

Comment:
Can you see if this will read better if the check was moved to `handleRestore` in tm_init.go?

Reply:
Better to keep it here so that calls from other call-sites also benefit from this change.